frontend/src/lib/api.ts

```diff
@@ -213,17 +214,21 @@ const api = {
             trailingEntry: PluginLogEntry | null = null,
             leadingEntry: PluginLogEntry | null = null
         ): Promise<PluginLogEntry[]> {
-            const type_filters =
-                typeFilters && typeFilters.length > 0 ? `&type_filter=${typeFilters.join('&type_filter=')}` : ''
-            const search = searchTerm ? `&search=${searchTerm}` : ''
-            const before = trailingEntry ? '&before=' + trailingEntry.timestamp : ''
-            const after = leadingEntry ? '&after=' + leadingEntry.timestamp : ''
-            const queryParams = `?limit=${LOGS_PORTION_LIMIT}${before}${after}${search}${type_filters}`
+            const params = toParams(
+                {
+                    limit: LOGS_PORTION_LIMIT,
+                    type_filter: typeFilters,
+                    search: searchTerm || '',
```

Comment:
`searchTerm || undefined`

Reply:
I _love_ a chain of map, reduce, and filter :)

> I think this if would look cleaner in the next, map block with a nested map to be honest, skipping this entire function.

I started as one function but it was hard to see what was the array fangling and what was the query parameter formatting

Added a comment to (try to) explain why the reduce has the branch

Reply:
I love the elegance and purity in this approach, yet elegance isn't strongly correlated to maintainability. The comment does the trick though. You normally trust the code to do what it says, so clarity improved here a lot 📈